### ISSUE TITLE
Allowing `<address>` inside `<div>`

### ISSUE THREAD

**Issue by StefanDumont** [TEIC/TEI#2460]
A discussion at a workshop on encoding correspondence resulted in the article [Addresses](https://encoding-correspondence.bbaw.de/v1/addresses.html) by Marjam Trautmann, in which several problems with `<address>` are being discussed.

One of the problems that Trautmann states:

> The `<address>` element is not part of the content model of `<div>`. This means that it cannot appear directly inside a `<div>` and this also means that it is not allowed before model.divTopPart or after model.divBottomPart elements, e.g. `<opener>` or after `<closer>`. So, if the encoder wants to model the addresses of recipient(s) and sender(s) as a separate structural block within the `<text>` element and wants to put the addresses before the `<opener>` or after `<closer>` of a letter, he or she needs to construct an elaborate data model with the addresses and the `<opener>`/`<closer>` encoded in several `<div>` elements. 

(see the [article](https://encoding-correspondence.bbaw.de/v1/addresses.html#c-3) for details)

> To avoid such an unnecessary elaborate encoding, it would be convenient to be able to use `<address>` directly inside a `<div>`. Could a possible solution be to assign the element `<address>` to a suitable additional class, like model.divTopPart and model.divBottomPart?

**Comment by StefanDumont** [comment]
In an [open annotation](https://hyp.is/_IEfHphuEeuDLAcrxdO2dQ/encoding-correspondence.bbaw.de/v1/addresses.html) on Trautmann's article, Jakub Šimek suggests that `<address>` could be set even more flexibly:

> I strongly support this proposal and would like to push it even further, allowing for the use of `<address>` not only directly inside of `<div>` but also directly in `<body>`, `<front>` or `<back>`, making it a "chunk level" element of its own. While grouping several addresses inside a `<div>` could be useful, using `<div>` should not be required for one single address.

**Comment by hcayless** [comment]
My question about this is whether there's anything unsatisfactory about wrapping the `<address>` in an `<ab>`? My instinct is that an address doesn't intrinsically have any rendering / structural semantics, which I think we'd be assuming by making it a direct child of `<div>`.

**Comment by sabineseifert** [comment]
Putting the address in `<ab>` does not solve the problem of a following `<opener>` and/or a preceding `<closer>`. To be valid, everything needs to be put in a `<div>` each which is considered as rather elaborate encoding.

**Comment by lb42** [comment]
You could of course put the address inside an opener or closer. Which could even be typed !

**Comment by sabineseifert** [comment]
Yes, that was in fact the proposal in the [article (issue 3)](https://encoding-correspondence.bbaw.de/v1/addresses.html#c-4). 

We then would need to adapt the definitions of both `<opener>` and `<closer>` a litte bit. 
([`<opener>`](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-opener.html): "combines date line, author, salutation and similar phrases that are used at the beginning of a section, especially in a letter".
[`<closer>`](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-closer.html):  contains "greeting formulas, date lines and similar phrases that appear at the end of a section, especially in a letter".)

In an [open annotation](https://hyp.is/Ko3EQphwEeuiED_6qN_IKA/encoding-correspondence.bbaw.de/v1/addresses.html) on the article, Jakub Šimek argues:
"In my opinion this would stretch the semantics of `<opener>` and `<closer>` too much. The distinction should be clear between what is part of the letter text and what isn't. The addresses, typically very clearly separated from the letter body as intended for the mail logistics, are not really part of the letter text. Some encoders (like me) could even rather put the transcribed addresses inside of `<front>` than of `<body>`, arguing that correspondence addresses are not part of the textual _body_."

**Comment by nquenouille** [comment]
Dear Stefan,
I am totally with you. We have documents (for example, copies of letters in a bundle of documents) that contain `<closer>`, `<postscript>` and an `<address>` on the same page, and it would be much trouble to encode the address separately here as Marjam Trautmann has to do it. So, indeed, I would be very happy to see the `<address>`-Tag put into a `<div>`-Tag.

**Comment by ebeshero** [comment]
When figuring out how to encode addresses for correspondence in my own projects (eg Digital Mitford) we did find, with Jakub Šimek here, that these blocks of directional instructions for mailing simply did not fit with the semantics of `<closer>` or `<opener>`. They are not really to do with the reading content directed to the recipient, so setting addresses inside this se elements seemed awkward to us too. In my 19c correspondence projects the address sits on a separate leaf and really does make a block on the page. Since we document postal marks / ink stamps about delivery in the TEI header in our project, I suppose it might make sense to include the mailing address there as well, though we didn't try that. Instead we went with some awkwardness of tucking them into the `<closer>` which felt a bit wrong. 

At the very least, I think addresses do make a "block" of information in the source document for correspondence. They are designed to be read by delivery  agents and should not just be considered part of the letter message itself. (In my early 19c project, letters were written out in sheets of paper, then folded several times, with a little panel exposed on the back of the last sheet for the address to be written out. That little panel with the address lines would be marked and stamped by the postal service people en route.)

I have only ever seen addresses be written out in block form with constituent lines and spacing (or folding or enveloping) to set them apart from other parts of a document. So it makes very good sense to me to set an `<address>` in a `<div>`.

**Comment by sabineseifert** [comment]
Subgroup at F2F in Buenos Aires 2024-10:
We like permitting `<address>` inside `<div>` under certain controlled circumstances. But there may be problems with how to add it to `<div>` if we choose to do so with `model.common`.
We can try adding `model.common` to `model.frontPart` (which is used by `<back>` and `<front>`).

**Comment by sydb** [comment]
Council (at its meeting today) makes this GO for assignees to create a branch, add `<address>` to model.divTopPart and model.divBottomPart, and test to see if it works. If it works, create PR.

**Comment by torstenroeder** [comment]
@bleekere added `address` successfully to both `model.divTopPart` and `model.divBottomPart` 🥳

**Comment by trishaoconnor** [comment]
Thank you very much for your work on this ticket. PR #2790 passes all tests locally and on GitHub so I've reviewed, approved, and merged the PR into dev. Closing this ticket as completed.

**Comment by sydb** [comment]
I am confused as to how this could have passed tests for @trishaoconnor and on GitHub, as there is now an ambiguity in the content model of `<post>` that I am reasonably confident is caused by [this](#2790) change. I am concerned that we may have a bug tin the tests.

In any case, the problem is that `<address>` element was already a member of model.phrase before the changes on the branch & PR, and the content model of `<post>` makes no attempt to restrict the model.divBottom stuff to the bottom. Thus XSD (and DTD) is scared that a parser which came across a post/address would not know if the `<address>` came from model.divBottom or model.phrase.)

**Comment by sydb** [comment]
This was fixed in #2818.
